File: app/src/main/java/de/eloc/eloc_control_panel/models/ElocMarker.java
package de.eloc.eloc_control_panel.models;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.android.gms.maps.model.LatLng;
import com.google.maps.android.clustering.ClusterItem;

public class ElocMarker implements ClusterItem {
    private final String title;
    private final String snippet;
    private final LatLng position;

    public ElocMarker(String title, String snippet, LatLng position) {
        this.title = title;
        this.snippet = snippet;
        this.position = position;
    }

    @NonNull
    @Override
    public LatLng getPosition() {
        return position;
    }

    @Nullable
    @Override
    public String getTitle() {
        return title;
    }

    @Nullable
    @Override
    public String getSnippet() {
        return snippet;
    }

    @Nullable
    @Override
    public Float getZIndex() {
        return 0f;
    }
}

File: app/src/main/java/de/eloc/eloc_control_panel/data/ElocMarker.kt
package de.eloc.eloc_control_panel.data

import com.google.android.gms.maps.model.LatLng
import com.google.maps.android.clustering.ClusterItem

class ElocMarker(
    private val title: String,
    private val snippet: String,
    private val position: LatLng
) : ClusterItem {

    override fun getPosition(): LatLng = position

    override fun getTitle(): String = title

    override fun getSnippet(): String = snippet

    override fun getZIndex(): Float = 0f
}
